### PR TITLE
Add post_up support to cumulus_interface

### PR DESCRIPTION
**Pull request by floored1585**
This adds the ability to add post_up lines to interface configuration when using cumulus_interface.  Let me know if you see anything that needs to be tweaked.

### CODE REVIEW

**Comment by LeslieCarr** [comment]
Thanks for the submission!  In a good stroke of timing luck, today is my "play with modules" day so i'll double check this later today (barring any emergencies)!

**Comment by LeslieCarr** [comment]
I shouldn't have mentioned barring any emergencies ;)  I promise I will test this early next week! Thank you!

**Comment by floored1585** [comment]
Haha, okay, no problem!

**Comment by floored1585** [comment]
Not sure why the build failed, maybe re-run it?  The tests are all passing on my side.

File: cookbooks/cumulus/providers/interface.rb [hunk 1]
```diff
@@ -40,6 +41,8 @@ def whyrun_supported?
   ipv6 = new_resource.ipv6
   address = ipv4 + ipv6
 
+  post_up = new_resource.post_up
```
Comment (Vanders) [inline cookbooks/cumulus/providers/interface.rb]: post_up is being set twice (which is what is triggering the build failure; Rubocop has picked it up). Could you remove this assignment?

**Comment by floored1585** [comment]
Oops, yep.  Fixed.

**Comment by skamithi** [comment]
hello @floored1585 Would like to have a post_down feature included with this PR. If you can code that up, and add to this pull request that would be much appreciated. Will start working on updating the ansible and puppet modules next week with  post_up and post_down features. Thanks.

**Comment by eric-cumulus** [comment]
Yea, we actually need this for Vx too, "post-up ip link set $IFACE promisc
on" is required for any interface config for swp ports involved in a bond
in Vx.

On Fri, Oct 23, 2015 at 9:33 AM, Stanley Karunditu <notifications@github.com
> wrote:

> hello @floored1585 <https://github.com/floored1585> Would like to have a
> post_down feature included with this PR. If you can code that up, and add
> to this pull request that would be much appreciated. Will start working on
> updating the ansible and puppet modules next week with post_up and
> post_down features. Thanks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CumulusNetworks/cumulus-linux-chef-modules/pull/18#issuecomment-150573874>
> .
>



-- 
Eric C. Pulvino <https://www.linkedin.com/in/pulvino> | Cumulus Networks
<http://cumulusnetworks.com> | Sr Consultant, Enablement |
eric@cumulusnetworks.com | 919.592.5995 <19195925995>

**Comment by floored1585** [comment]
Yeah I can add pre_down to this.  I also made some changes in a separate branch to fix ifquery comparison for accurate 'notifies' behavior (right now, at least on 2.5.4, the Interfaces provider _always_ thinks an interface needs to be updated, even though it doesn't end up changing the file).  Do you want those fixes as part of this pull request or a separate one?

**Comment by floored1585** [comment]
Do the current functionalities and tests look OK for this pull req?  Looking forward to getting this and the other (#19) merged so we can switch back to using the official cookbook.  Thanks!

**Comment by skamithi** [comment]
@floored1585  thanks for the PRs.

**Pull request merged**